roxctl/netpol/connectivity/diff/netpol_test.go

```diff
@@ -50,18 +44,92 @@ func (d *diffAnalyzeNetpolTestSuite) TestDiffAnalyzeNetpol() {
 			inputFolderPath2:      "",
 			expectedValidateError: errox.InvalidArgs,
 		},
+		{
+			name:                  "Valid inputs should write output to a provided output file",
+			inputFolderPath1:      "testdata/netpol-analysis-example-minimal",
+			inputFolderPath2:      "testdata/netpol-diff-example-minimal",
+			expectedValidateError: nil,
+			outFile:               outFileTxt,
+			runCommand:            true, // to write the output - critic for next tests
+			removeOutputPath:      false,
+		},
+		{
+			name:                  "Existing output file input without using remove flag should return error that the file already exists",
+			inputFolderPath1:      "testdata/netpol-analysis-example-minimal",
+			inputFolderPath2:      "testdata/netpol-diff-example-minimal",
+			expectedValidateError: errox.AlreadyExists,
+			outFile:               outFileTxt,
+			removeOutputPath:      false,
+		},
+		{
+			name:                  "Existing output file input with using remove flag should override existing file",
+			inputFolderPath1:      "testdata/netpol-analysis-example-minimal",
+			inputFolderPath2:      "testdata/netpol-diff-example-minimal",
+			expectedValidateError: nil,
+			runCommand:            true,
+			outFile:               outFileTxt,
+			removeOutputPath:      true,
+		},
+	}
+
+	for _, tt := range cases {
+		tt := tt
+		d.Run(tt.name, func() {
+			env, _, _ := mocks.NewEnvWithConn(nil, d.T())
+			diffNetpolCmd := diffNetpolCommand{
+				inputFolderPath1: tt.inputFolderPath1,
+				inputFolderPath2: tt.inputFolderPath2,
+				outputFilePath:   tt.outFile,
+				removeOutputPath: tt.removeOutputPath,
+				env:              env,
+			}
+
+			analyzer, err := diffNetpolCmd.construct()
+			d.NoError(err)
+
+			err = diffNetpolCmd.validate()
+			if tt.expectedValidateError != nil {
+				d.Require().Error(err)
+				d.ErrorIs(err, tt.expectedValidateError)
+				return
+			}
+			d.NoError(err)
+
+			if tt.runCommand {
+				err := diffNetpolCmd.analyzeConnectivityDiff(analyzer)
```

Comment:
The goal of this test was to not run `analyzeConnectivityDiff` but focus only on  `validate`.

Reply:
fixed
moved the test with overriding the existing output file to the `TestAnalyzeDiffCommand` (as it needs running the command)